### PR TITLE
[terra-table] Provide screenreader value for row selection column header

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The Table component was updated such that a screenreader value was associated with the row selection column header.  The component was also updated to properly read the header text and error indicator in the proper order.  It now also properly reads the sort indicator and associated direction changes on all screenreaders.

**Why it was changed:**
The change was made to improve the accessibility of the Table component.

### Testing

This change was tested using:

- [ ] WDIO
- [X] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [X] Accessibility review
- [X] Functional review

### Additional Details
N/A

**This PR resolves:**

UXPLATFORM-9867 
UXPLATFORM-9865
UXPLATFORM-9685

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by eawww — APPROVED** [review]

**Comment by mjpalazzo** [comment]
@cm9361 I did the following validation with Edge/JAWS. I have a question for the row selection example:
Table Column States example - Pressed 'T' to navigate to table. JAWS announced, "Table with 6 columns and 3 rows, Table column 1 row 1, Patient, errors associated with column, and sorted ascending button." **This is now working as required.**
Row Selection example, Row selection mode enabled - Pressed 'T' to navigate to table. JAWS announced, "Table with 4 columns and 6 rows, Table with row selection, column 1 row 1, row selection button." **JAWS now announces something for the checkbox column header, but do we want it to announce "button" or "row selection column header"?**
Pressed Down Arrow to navigate the header cells until I reached the first check box cell. JAWS announced, "row 3 checkbox not checked." **This is row 2. JAWS announced the wrong row number**
Pressed Space to check the checkbox. JAWS announced Space, **row 3**, Fleck, Arthur, Fleck, Arthur selection, column 1 **row 2** checkbox checked. To clear checkbox, press space bar. **Row 2** selected. **The problem with the incorrect row index is present. Is that being addressed in a different PR?**

**Comment by cm9361** [comment]
> @cm9361 I did the following validation with Edge/JAWS. I have a question for the row selection example: Table Column States example - Pressed 'T' to navigate to table. JAWS announced, "Table with 6 columns and 3 rows, Table column 1 row 1, Patient, errors associated with column, and sorted ascending button." **This is now working as required.** Row Selection example, Row selection mode enabled - Pressed 'T' to navigate to table. JAWS announced, "Table with 4 columns and 6 rows, Table with row selection, column 1 row 1, row selection button." **JAWS now announces something for the checkbox column header, but do we want it to announce "button" or "row selection column header"?** Pressed Down Arrow to navigate the header cells until I reached the first check box cell. JAWS announced, "row 3 checkbox not checked." **This is row 2. JAWS announced the wrong row number** Pressed Space to check the checkbox. JAWS announced Space, **row 3**, Fleck, Arthur, Fleck, Arthur selection, column 1 **row 2** checkbox checked. To clear checkbox, press space bar. **Row 2** selected. **The problem with the incorrect row index is present. Is that being addressed in a different PR?**

Thank you for pointing out these issues.  There must have been a breaking change at some point.  We would need to resolve the issue in another PR.  Could you please log a JIRA?

File: packages/terra-table/src/subcomponents/ColumnHeaderCell.jsx [hunk 1]
```diff
@@ -62,6 +63,11 @@ const propTypes = {
    */
   isActive: PropTypes.bool,
 
+  /**
+   * Boolean value indicating whether or not the header cell is focused.
+   */
+  isDisplayVisible: PropTypes.bool,
+
```
Comment (sdadn) [inline packages/terra-table/src/subcomponents/ColumnHeaderCell.jsx:70]: Is the prop description correct? It seems like the value is used to show/hide the display text.

**Review by sdadn — APPROVED** [review]
Question about the prop description but code wise it looks good so approving it.

File: packages/terra-table/src/subcomponents/ColumnHeaderCell.jsx (hunk 1 above)
Comment (cm9361) [reply packages/terra-table/src/subcomponents/ColumnHeaderCell.jsx:70]: Yeah, I made a mistake here.  Ugh!

**Review by smason0 — APPROVED** [review]

**Pull request merged**